File: java/src/main/java/mx/iteso/RealUser.java

package mx.iteso;

/** RealUser implementation class. */
public class RealUser extends User {

    /**
     * Publish permission initially denied function.
     * @return boolean Permission.
     */
    public boolean publishPermission() {
        System.out.println("Publishing not allowed.");
        return false;
    }

    /**
     * Deletion permission initially denied function.
     * @return boolean Permission.
     */
    public boolean deletionPermission() {
        System.out.println("Deletion not allowed.");
        return false;
    }

    /**
     * Approval permission initially denied function.
     * @return boolean Permission.
     */
    public boolean approvalPermission() {
        System.out.println("Approval not allowed.");
        return false;
    }

    /**
     * Comment permission initially denied function.
     * @return boolean Permission.
     */
    public boolean commentPermission() {
        System.out.println("Commenting not allowed.");
        return false;
    }
}

File: python/src/main/python/RealUser.py
from User import *

class RealUser(User):

    def publishPermission(self):
        print("Publishing not allowed.")
        return False

    def deletionPermission(self):
        print("Deletion not allowed.")
        return False

    def approvalPermission(self):
        print("Approval not allowed.")
        return False

    def commentPermission(self):
        print("Commenting not allowed.")
        return False

